simple/watcher.go

```diff
@@ -0,0 +1,48 @@
+package simple
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-app-sdk/resource"
+)
+
+// Watcher is a struct that implements operator.ResourceWatcher and SyncWatcher, but takes no action on its own.
+// For each method in (Add, Update, Delete) the corresponding exported function field is called, if non-nil.
+type Watcher struct {
```

Comment:
What would be the use-case for this?

Reply:
Namely to be easily wrapped by `operator.OpinionatedWatcher` with a dedicated `Sync` method to map to the `OpinionatedWatcher` sync events.

Reply:
Sorry, my bad – I wasn't clear enough with my question. IIUC this is used to provide the `WatchKind` functionality, but I was wondering whether we even should provide it.

I suggest that we limit the simple use-case to reconciling operators only, WDYT?